src/cssLanguageService.ts

```diff
@@ -72,15 +80,34 @@ function createFacade(parser: Parser, completion: CSSCompletion, hover: CSSHover
 	};
 }
 
+function handleCustomData(options?: LanguageServiceOptions) {
+	if (options) {
+		if (options.customProperties) {
+			addProperties(options.customProperties);
+		}
+		if (options.customAtDirectives) {
+			addAtDirectives(options.customAtDirectives);
+		}
+		if (options.customPseudoClasses) {
+			addPseudoClasses(options.customPseudoClasses);
+		}
+		if (options.customPseudoElements) {
+			addPseudoElements(options.customPseudoElements);
+		}
+	}
+}
 
-export function getCSSLanguageService(): LanguageService {
+export function getCSSLanguageService(options?: LanguageServiceOptions): LanguageService {
+	handleCustomData(options);
```

Comment:
If we choose this API, we must store the properies/directives per LanguageService, not in global arrays. Users can create any number of language services, each with a different set of data.

I would suggest to make a new type `CSSPropertySet` (name to be debated) containing properties/directives/..., that is registered globaly with the `vscode-css-languageservice`. Our built-in properies (from MDN) should also be represented in one (or multiple) CSSPropertySets. 

In a first step, all language services just use that global list of propertySets. In a later step we allow to create a language service with a custom list of Sets. That language service would then only use that set. It would be good to already prepare our code to avoid accessing CSSPropertySet globaly.

Reply:
Yeah I was about to check with you. The same problem with HTML now. Do we make the language services stateful? I'll go with stateful then.